### ISSUE TITLE
Native Softmax

### ISSUE THREAD

**Issue by MikeInnes** [JuliaGPU/CUDA.jl#175]
Can be based on [PyTorch](https://github.com/pytorch/pytorch/blob/master/aten/src/THCUNN/SoftMaxCommon.cuh#L389)'s.

The API should be equivalent to:

```julia
function softmax(x::AbstractArray, axis=1)
    out = exp.(x .- maximum(x, axis))
    out ./= sum(out, axis)
    return out
end
```

**Comment by maleadt** [comment]
This is something for NNlib.jl.
